test/functional/test_framework/util.py

```diff
@@ -323,12 +323,14 @@ def get_auth_cookie(datadir):
                 if line.startswith("rpcpassword="):
                     assert password is None  # Ensure that there is only one rpcpassword line
                     password = line.split("=")[1].strip("\n")
-    if os.path.isfile(os.path.join(datadir, "regtest", ".cookie")) and os.access(os.path.join(datadir, "regtest", ".cookie"), os.R_OK):
+    try:
         with open(os.path.join(datadir, "regtest", ".cookie"), 'r', encoding="ascii") as f:
             userpass = f.read()
             split_userpass = userpass.split(':')
             user = split_userpass[0]
             password = split_userpass[1]
+    except FileNotFoundError:
```

Comment:
```suggestion
    except OSError:
```

To also catch PermissionError, to not regress on 2753285be72dd051cf2987ca5c80d0b149914020?

See https://docs.python.org/3/library/exceptions.html#os-exceptions